### ISSUE TITLE
Train & deploy predictive content virality scoring model (pre-publish quality gate)

### ISSUE THREAD

**Issue by BestGaaS220** [kushin77/news-feed-engine#51]
## Task Description

Train and deploy a machine learning model that predicts the virality and performance score of a piece of content before it is published. This gives the content pipeline a pre-publish quality gate that filters out weak content and prioritizes high-potential posts.

## Model Architecture

```python
# Feature engineering from content attributes
CONTENT_FEATURES = [
    # Text features
    "hook_type",                    # pattern interrupt, question, stat, story
    "caption_length",               # chars
    "hashtag_count",
    "emoji_density",
    "sentiment_score",              # positive/negative valence
    "reading_level",                # Flesch-Kincaid
    "has_cta",                      # bool
    "cta_type",                     # follow, link, comment, share
    
    # Visual features (images/thumbnails)
    "image_brightness",
    "face_present",                 # bool
    "face_expression",              # happy, surprised, neutral
    "text_overlay_present",         # bool
    "brand_color_match",            # cosine similarity to brand palette
    
    # Timing features
    "publish_hour",
    "publish_day_of_week",
    "days_since_last_post",
    
    # Trend relevance
    "trend_relevance_score",        # from Researcher Agent
    "competitor_saturation",        # how many similar posts exist
]

TARGET = "top_quartile_engagement"  # binary: is this post in top 25%?

# XGBoost classifier trained on historical posts
model = XGBClassifier(n_estimators=500, learning_rate=0.05, max_depth=6)
```

## Training Pipeline

- [ ] Feature extraction pipeline for all historical posts (≥6 months of data)
- [ ] Label generation (top quartile engagement per platform)
- [ ] Model training: XGBoost with cross-validation
- [ ] Model evaluation: AUC-ROC, precision/recall, calibration curve
- [ ] MLflow experiment tracking (model versions, metrics, parameters)
- [ ] Model serving: FastAPI endpoint (`POST /predict/virality-score`)

## Integration

- Content Creator Agent calls scoring API before finalizing variants
- Low-score variants (<0.4) recycled with prompt variations
- High-score variants (>0.75) fast-tracked to publishing queue

## Acceptance Criteria

- [ ] Model AUC-ROC ≥ 0.72 on held-out test set
- [ ] Prediction latency < 100ms per post
- [ ] Model improves predictive accuracy as post volume grows (online learning)
- [ ] MLflow tracking operational with ≥3 model versions checkpointed
- [ ] Scoring API deployed and integrated with Content Creator Agent (#45)
- [ ] Top-scored posts outperform low-scored posts by ≥40% on median engagement

**Effort:** 5 days | **Priority:** P1 | **Parent Epic:** #32 | **Deps:** #44 (data collection running)
